pkg/app/piped/controller/scheduler.go

```diff
@@ -687,6 +705,25 @@ func (s *scheduler) reportDeploymentCompleted(ctx context.Context, status model.
 	return err
 }
 
+func (s *scheduler) shouldSkipStage(ctx context.Context, in executor.Input) (skip bool, err error) {
+	stageConfig := in.StageConfig
+	var skipOptions config.SkipOptions
+	switch stageConfig.Name {
+	case model.StageAnalysis:
+		skipOptions = stageConfig.AnalysisStageOptions.SkipOn
+	case model.StageWait:
+		skipOptions = stageConfig.WaitStageOptions.SkipOn
+	case model.StageWaitApproval:
+		skipOptions = stageConfig.WaitApprovalStageOptions.SkipOn
+	case model.StageScriptRun:
+		skipOptions = stageConfig.ScriptRunStageOptions.SkipOn
+	default:
+		return false, nil
```

Comment:
I think the current one is enough because
- `StageConfig.GetSkipConfig()` will be used only in `skipstage` package 
- adding the func to each stage model seems redundant

WDYT?

Reply:
OK, I got it 👍